src/sdk/bucketspace_fs.js

```diff
@@ -314,14 +316,30 @@ class BucketSpaceFS extends BucketSpaceSimpleFS {
     }
 
 
-    new_bucket_defaults(account, { name, tag, lock_enabled, force_md5_etag }, create_uls, bucket_storage_path) {
+    async new_bucket_defaults(account, { name, tag, lock_enabled, force_md5_etag }, create_uls, bucket_storage_path) {
+        // if is it root account then it is the owner
+        // else (it is an IAM account) the owner is it's root account
+        const is_account_root_account = manage_nsfs_cli_utils.check_root_account(account);
+        const owner_account_id = is_account_root_account ? account._id : account.owner;
+        let owner_account_config;
+        // in case it is IAM account we need the name of account (we have only the id in owner property)
+        if (!is_account_root_account) {
+            // global_config specific for this function
+            const global_config = {
+                config_root_backend: this.fs_context.fs_backend,
+                accounts_dir: this.accounts_dir,
+            };
+            owner_account_config = await nc_utils.get_account_by_id(global_config, account.owner);
```

Comment:
this is bad, we are going to iterate all accounts just to find the root account name here?? this is bad, can't we just avoid saving the root account name in the bucket??

Reply:
The whole changes in the last commit were because I didn't have the root account name and only the id.
Those properties are planned to be removed in next version (`system_owner`, `bucket_owner`).
I can add to the schema a non-required property with the root account name on IAM creation and in the next version we can remove it (like we remove `bucket_owner`). WDYT?

Reply:
so why not keep using bucket_owner field for now? isn't that the account name?

Reply:
In this current fix, we set the `system_owner` and `bucket_owner` with the root account name.
As mentioned I don't have the account name, so at first suggestion I iterate and searched it (on all the accounts), in this suggestion I added this property in the schema.

I don't want to use the current account name because I want to validate for 2 IAM accounts that the can access the buckets of the same root account.